### PR TITLE
Pass compile.sh the number of CPUs to multi-thread

### PR DESCRIPTION
**Pull request by mclare**
Speeds up compile by using all available cores. Tested on Fedora and OSX.

### CODE REVIEW

**Comment by mclare** [comment]
bump ;)

File: installer.sh [hunk 1]
```diff
@@ -303,7 +303,15 @@ else
 		set -e
 		echo "[3/3] No prebuilt PHP found, compiling PHP automatically. This might take a while."
 		echo
-		exec "./compile.sh"
+		logical_cpu_count=$([ "$(uname -s)" == "Darwin" ] && sysctl -n hw.logicalcpu_max || nproc --all) #Get available CPUs to pass to compile.sh and speed up compile
```
Comment (SOF3) [inline installer.sh:306]: What about using `command -v nproc`? Not all operating systems have `nproc`.

Comment (dktapps) [reply installer.sh:306]: which ones don't?

Comment (mclare) [reply installer.sh:306]: OSX is certainly an OS without nproc, which is addressed with sysctl -n hw.logicalcpu_max, but I can't answer which others don't. Perhaps other BSDs? I've only tested on OSX (arm64) and Ubuntu 16 (intel) & 20 (arm/intel) and Fedora (arm64). 

... I think it works well, but maybe allow logical_cpu_count to be manually set too?

Comment (dktapps) [reply installer.sh:306]: I suppose it doesn't matter too much, as long as it gracefully falls back to 1 on failure.

**Review by SOF3 — APPROVED** [review]
tbh I would rather see this change in compile.sh, but I guess those who want to explicitly compile themselves should know well enough.

**Comment by dktapps** [comment]
A warning was recently added to compile.sh for compiling in single-thread mode. We previously discussed automatically using more cores, but there were reasons not to (other process starvation).

**Pull request merged**